dlpy/tests/test_gpu.py

```diff
@@ -0,0 +1,539 @@
+    Please locate the images.sashdat file under the datasources to the DLPY_DATA_DIR.
+    '''
+    server_type = None
+    s = None
+    server_sep = '/'
+    data_dir = None
+    data_dir_local = None
+
+    def setUp(self):
+        swat.reset_option()
+        swat.options.cas.print_messages = False
+        swat.options.interactive_mode = False
+
+        self.s = swat.CAS()
+        self.server_type = tm.get_cas_host_type(self.s)
+        self.server_sep = '\\'
+        if self.server_type.startswith("lin") or self.server_type.startswith("osx"):
+            self.server_sep = '/'
+
+        if 'DLPY_DATA_DIR' in os.environ:
+            self.data_dir = os.environ.get('DLPY_DATA_DIR')
+            if self.data_dir.endswith(self.server_sep):
+                self.data_dir = self.data_dir[:-1]
+            self.data_dir += self.server_sep
+
+        if 'DLPY_DATA_DIR_LOCAL' in os.environ:
+            self.data_dir_local = os.environ.get('DLPY_DATA_DIR_LOCAL')
+            if self.data_dir_local.endswith(self.server_sep):
+                self.data_dir_local = self.data_dir_local[:-1]
+            self.data_dir_local += self.server_sep
+
+    def CreateSimpleCNN1(self):
+        model1 = Sequential(self.s, model_table='Simple_CNN1')
+        model1.add(InputLayer(3, 224, 224))
+        model1.add(Conv2d(8, 7))
+        model1.add(Pooling(2))
+        model1.add(Conv2d(8, 7))
+        model1.add(Pooling(2))
+        model1.add(Dense(16))
+        model1.add(OutputLayer(act='softmax', n=2))        
+        return model1
+
+    def LoadEEE(self):
+        if self.data_dir is None:
+            unittest.TestCase.skipTest(self, "DLPY_DATA_DIR is not set in the environment variables")
+
+        caslib, path, tmp_caslib = caslibify(self.s, path=self.data_dir+'images.sashdat', task='load')
+
+        self.s.table.loadtable(caslib=caslib,
+                               casout={'name': 'eee', 'replace': True},
+                               path=path)
+        return caslib, tmp_caslib, 'eee', '_image_', '_label_'
+
+    def test_gpu01(self):
+        '''
+        Enable GPU with all default values in Model.fit() and Model.predict().
+            Gpu parameters object.
+
+            Parameters
+            ----------
+            devices : list-of-ints, optional
+                Specifies a list of GPU devices to be used.
+            use_tensor_rt : bool, optional
+                Enables using TensorRT for fast inference.
+                Default: False.
+            precision : string, optional
+                Specifies the experimental option to incorporate lower computational
+                precision in forward-backward computations to potentially engage tensor cores.
+                Valid Values: FP32, FP16
+                Default: FP32
+            use_exclusive : bool, optional
+                Specifies exclusive use of GPU devices.
+                Default: False
+
+            Returns
+            -------
+            :class:`Gpu`
```

Comment:
The docstring for a unit test should state the purpose of the test and indicate code coverage.  All unit tests are self-contained, so there are no parameters.